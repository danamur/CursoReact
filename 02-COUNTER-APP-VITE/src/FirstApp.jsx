export const FirstApp = () => {
  return (
    <>
        <h1>Daniel Namur</h1>
        <p>Desarrollador Full Stack</p>
    </>
  )
}