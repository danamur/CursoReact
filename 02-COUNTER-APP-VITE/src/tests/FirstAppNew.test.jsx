import { render, screen } from '@testing-library/react'
import { FirstApp } from '../FirstApp';

describe('Pruebas en <FirstApp />', () => {

    const title = 'Bienvenido, Dwight';
    const subTitle = "Soy un subtitulo";

    test('pruebas en <FirstApp /> ', () => {

        const { container } = render(<FirstApp title={title}/>);
        expect(container).toMatchSnapshot();

    });

    test('Debe de mostrar el mensaje "Bienvenido, Dwight', () => {
       
        render( <FirstApp title={title} />);
        expect(screen.getByText(title)).toBeTruthy();

    });

    test('Debe de mostrar el titulo en un h1', () => {

        render( <FirstApp title={title} />);
        expect(screen.getByRole('heading', { level: 1}).innerHTML).toContain(title);

    });

    test('Debe de mostrar el subtitulo enviado por props', () => {

        render( <FirstApp title={title} />);
        expect(screen.getAllByText(subTitle).length).toBe(1);

    });

});