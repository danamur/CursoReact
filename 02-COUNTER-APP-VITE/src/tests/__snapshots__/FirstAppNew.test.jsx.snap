// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> pruebas en <FirstApp />  1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Bienvenido, Dwight
  </h1>
  <p>
    Soy un subtitulo
  </p>
  <p>
    Daniel Namur
  </p>
</div>
`;
